### DISCUSSION TITLE
Force "Consumer Cancel Notification"

### DISCUSSION THREAD

**Discussion by andiwaldi** [rabbitmq/rabbitmq-server#10016] [Questions] [answered]
Is there anyway for a rabbit-mq server to force a broker to send out "basic.cancel" to a consumer?
https://www.rabbitmq.com/consumer-cancel.html

I would like to test if your Application is handling them correctly, but so far I have found no Way to actually send out a "basic.cancel" from the Broker

**Comment by michaelklishin** [comment, marked as answer]
@andiwaldi from the linked doc guide:

```
Other events, such as the queue being deleted, or in a clustered scenario, the node on which the queue is located failing, will cause the consumption to be cancelled
```

Deleting the queue is the easiest way to test because in modern RabbitMQ versions, a queue replica migration will be handled by RabbitMQ itself (it will re-register the consumer with the new leader without notifying the client).
